tests/text/test_eed.py

```diff
@@ -0,0 +1,149 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+
+import numpy as np
+import pytest
+from torch import Tensor, tensor
+
+from tests.text.helpers import INPUT_ORDER, TextTester
+from tests.text.inputs import (
+    _inputs_multiple_references,
+    _inputs_single_reference,
+    _inputs_single_sentence_multiple_references,
+)
+from torchmetrics.functional.text.eed import eed
+from torchmetrics.text.eed import EED
+
+
+def rwth_manual_metric(targets, preds) -> Tensor:
+    """The results were obtained w.r.t.
+
+    the examples defined in `tests.text.inputs` with the script from https://github.com/rwth-i6/ExtendedEditDistance.
+    """
+    ans_1 = tensor(0.24248056001808083)
+    ans_2 = tensor(0.19152276295133436)
+
+    HYPOTHESIS_A = "It is a guide to action which ensures that the military always obeys the commands of the party"
+
+    if len(preds) == 4:
+        return (ans_1 + ans_2) / 2
+    if HYPOTHESIS_A in preds:
+        return ans_1
+    return ans_2
+
+
+@pytest.mark.parametrize(
+    ["preds", "targets"],
+    [(_inputs_single_reference.preds, _inputs_single_reference.targets)],
+)
+class TestEED(TextTester):
+    # @pytest.mark.parametrize("ddp", [False, True])
+    # @pytest.mark.parametrize("dist_sync_on_step", [False, True])
+    def test_eed_class(self, preds, targets):
+        rwth_metric = partial(rwth_manual_metric)
+        self.run_class_metric_test(
+            ddp=False,
+            preds=preds,
+            targets=targets,
+            metric_class=EED,
+            sk_metric=rwth_metric,
+            dist_sync_on_step=True,
+            input_order=INPUT_ORDER.TARGETS_FIRST,
+        )
+
+    def test_eed_functional(self, preds, targets):
+        rwth_metric = partial(rwth_manual_metric)
+        self.run_functional_metric_test(
+            preds,
+            targets,
+            metric_functional=eed,
+            sk_metric=rwth_metric,
+            input_order=INPUT_ORDER.TARGETS_FIRST,
+        )
```

Comment:
```suggestion
            sk_metric=rwth_metric,
        )
```

as there's gonna be the `preds, target` order.

Reply:
adding to next commit